cache/cache.go

```diff
@@ -61,76 +63,97 @@ func New[T any](expireAfter time.Duration) *Cache[T] {
 		panic("expireAfter duration must be positive")
 	}
 
-	return &Cache[T]{
-		data:        make(map[string]item[T]),
+	c := &Cache[T]{
+		data:        make(map[string]*cacheListItem[T]),
 		expireAfter: expireAfter,
+		stopCh:      make(chan struct{}),
 	}
-}
-
-// Removes an item by name and expiry time when the purge was scheduled.
-// If there is a race, and the item has been refreshed, it will not be purged.
-func (c *Cache[T]) purgeExpired(name string, expectedExpiryTime int64) {
-	c.mu.Lock()
-	defer c.mu.Unlock()
 
-	if item, ok := c.data[name]; ok && item.expiresAt == expectedExpiryTime {
-		// found, and the expiry time is still the same as when the purge was requested.
-		delete(c.data, name)
+	// Start the sweep, with a minimum sweep of 50ms.
+	sweep := expireAfter / 4.0
+	if min := 50 * time.Millisecond; sweep < min {
```

Comment:
Same question about why 50 ms. I would expect, for example, a cache of public keys could get away with a longer time without issue.

Reply:
I expect most caches to be O(minutes) or O(hours). This is guarding against a case where someone makes a cache of 1ns and then we expect the sweep to happen more often than is physically possible. Even for small values (e.g. 10ms), we want to avoid pegging the CPU. The cache still checks if items are expired before returning them, so there's a guard against returning an item that should have been expired.

Reply:
I recommend adding the "why" as a code comment, so someone years later knows the context and can make a change if needed.

Reply:
Commented inline